ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -92,15 +104,57 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDefinitions = append(containerDefinitions, containerDef)
 	}
 
+	var ecsFields *ECSFields
+	if ecsFieldsFileName != "" {
+		data, err := readECSFields(ecsFieldsFileName)
+		ecsFields = data
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var networkMode string
+	if ecsFields != nil {
+		networkMode = ecsFields.TaskDefinition.NetworkMode
+		// The task-role-arn flag should take precedence over a taskRoleArn value specified in ECS fields file
+		if taskRoleArn == "" {
+			taskRoleArn = ecsFields.TaskDefinition.TaskRoleArn
+		}
+	}
+
+	if networkMode == "" {
+		networkMode = "bridge"
```

Comment:
Wouldn't the API set this to bridge? why do we need to manually set it?

Reply:
It does if the backend receives nil, but since we're passing in the NetworkMode field in `GetOrCreateTaskDefinition` (see entity_helper.go), this was getting set as an empty string, which was causing the integration test to fail:

```
~/sandbox/test_fields$ ecs-cli compose --file docker-compose.yml up --cluster test-ecs-fields
WARN[0000] Skipping unsupported YAML option...           option name=networks
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=wordpress
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=mysql
ERRO[0000] Error registering task definition             error="ClientException:  is not a valid network mode. Must be one of [bridge,host,none,awsvpc]\n\tstatus code: 400, request id: 222348ab-9cc4-11e7-9360-078fc3ccadc0" family="ecscompose-test_fields"
ERRO[0000] Create task definition failed                 error="ClientException:  is not a valid network mode. Must be one of [bridge,host,none,awsvpc]\n\tstatus code: 400, request id: 222348ab-9cc4-11e7-9360-078fc3ccadc0"
FATA[0000] ClientException:  is not a valid network mode. Must be one of [bridge,host,none,awsvpc]
	status code: 400, request id: 222348ab-9cc4-11e7-9360-078fc3ccadc0 
```

Reply:
Setting these modes will make this Linux-only.  If you can make this a `*string` and only pass it in when non-nil (or just check for `""` and use `nil` instead), that will let ECS do the right thing for Windows.  (Windows supports different network modes than Linux does, but we haven't modeled that fully in the task definition yet.)

Reply:
Good catch. Fixed, I think.